datafusion/execution/src/memory_pool/pool.rs

```diff
@@ -54,7 +58,44 @@ impl MemoryPool for UnboundedMemoryPool {
 #[derive(Debug)]
 pub struct GreedyMemoryPool {
     pool_size: usize,
-    used: AtomicUsize,
+    meta: Mutex<SimplePool>,
+}
+
+// Simple struct that tracks memory usage across multiple consumers.
+// Can be used in composition with other fields for providing more sophisticated memory pool
+// policies
+#[derive(Debug, Default)]
+struct SimplePool {
+    // Maps consumer names to the amount of memory they use.
+    pool_members: HashMap<String, usize>,
+    // The total memory usage of all consumers.
+    total_size: usize,
```

Comment:
I think it would help the code's readiblity to use call this `used` to match with the previous implementation

Reply:
✅